libs/processor_core.py

```diff
@@ -0,0 +1,93 @@
+from threading import Thread
+from queue import Queue
+from multiprocessing.managers import BaseManager
+import logging
+from share.commands import Commands
+from libs.distancing import Distancing as CvEngine
+
+logger = logging.getLogger(__name__)
+
+class QueueManager(BaseManager): pass
+
+
+class ProcessorCore:
+
+    def __init__(self, config):
+        self.config = config 
+        self._cmd_queue = Queue()
+        self._result_queue = Queue()
+        self._setup_queues()
+        self._engine = CvEngine(self.config)
+        self._tasks = {}
+       
+
+    def _setup_queues(self):
+        QueueManager.register('get_cmd_queue', callable=lambda: self._cmd_queue)
+        QueueManager.register('get_result_queue', callable=lambda: self._result_queue)
+        self._host = self.config.get_section_dict("CORE")["Host"]
+        self._queue_port = int(self.config.get_section_dict("CORE")["QueuePort"])
+        auth_key = self.config.get_section_dict("CORE")["QueueAuthKey"]
+        self._queue_manager = QueueManager(address=(self._host, self._queue_port), authkey=auth_key.encode('ascii'))
+        self._queue_manager.start()
+        self._cmd_queue = self._queue_manager.get_cmd_queue()
+        self._result_queue = self._queue_manager.get_result_queue()
+        logger.info("Core's queue has been initiated")
+
+       
+    def start(self):
+        logging.info("Starting processor core")
+        self._serve()
+        logging.info("processor core has been terminated.")
+        
+
+    def _serve(self):
+        while True:
+            logger.info("Core is listening for commands ... ")
+            cmd_code = self._cmd_queue.get()
+            logger.info("command received: " + str(cmd_code))
+            
+            if cmd_code == Commands.RESTART:
+                # Do everything necessary ... make sure all threads in tasks are stopped 
+                if Commands.PROCESS_VIDEO_CFG in self._tasks.keys():
+                    logger.warning("currently processing a video, stopping ...")
+                    self._engine.stop_process_video()
+                
+                # TODO: Be sure you have done proper action before this so all threads are stopped
```

Comment:
Is the idea to leave this TODO? Was it resolved by a commit?

Reply:
It just informs future developers to remember what needs to be done here if they are adding further features. Now the only task maybe running is the task of processing the video which his taken care of